### ISSUE TITLE
Array{T,n} vs Array(T,n)

### ISSUE THREAD

**Issue by timholy** [tonyhffong/Lint.jl#55]
A relevant example on [stackoverflow](http://stackoverflow.com/questions/27029110/julia-index-an-array-with-int64) is not caught:

``` julia
n = 32
A = Array{Int32, n}  # bug is here
for i in 1:n
    A[i] = i
end
```
